main.go

```diff
@@ -43,6 +45,92 @@ type configuration struct {
 	SseKey              string
 }
 
+var (
+	store = sessions.NewCookieStore([]byte(os.Getenv("SESSION_KEY")))
+)
+
+type formData struct {
+	ErrorMsg string
+}
+
+func getCredentials() (string, string) {
+	return os.Getenv("USER"), os.Getenv("PASSWORD")
+}
+
+func unauthorized(w http.ResponseWriter) {
+	w.Header().Set("WWW-Authenticate", `Basic realm="Restricted"`)
+	http.Error(w, "Unauthorized", http.StatusUnauthorized)
+}
+
+func SessionAuthMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path == "/login" {
+			next.ServeHTTP(w, r)
+			return
+		}
+
+		session, err := store.Get(r, "auth-session")
+		if err != nil {
+			http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+			return
```

Comment:
Out of curiosity, why did you feel passing `"Internal Server Error"` would be better than just passing `err.Error()`?